packages/grid/x-data-grid/src/components/base/GridBody.tsx

```diff
@@ -111,6 +111,13 @@ function GridBody(props: GridBodyProps) {
     };
   }, [apiRef]);
 
+  React.useEffect(() => {
+    // For JSDOM test environment that doesn't support ResizeObserver
+    if (typeof ResizeObserver === 'undefined' && process.env.NODE_ENV === 'test') {
+      apiRef.current.computeSizeAndPublishResizeEvent();
+    }
+  }, [apiRef]);
```

Comment:
> I suspect we didn't face this problem in our tests because of StrictMode.

Yes, I can confirm this.
We don't have a global ResizeObserver mock and strict mode is the only reason the tests pass without it since useLayoutEffect runs twice